From b64f8848efe610c52791587ec41301b73ec9165a Mon Sep 17 00:00:00 2001
From: Juan Quintela <quintela@redhat.com>
Date: Tue, 3 Aug 2021 11:53:07 +0200
Subject: [PATCH 07/39] Remove RHEL 7.4.0 machine types

RH-Author: quintela1 <quintela@redhat.com>
RH-MergeRequest: 30: Remove RHEL 7.x.0 machine types
RH-Commit: [5/6] a1940ac35591cebff52379f85656caf6a03328dd (juan.quintela/qemu-kvm)
RH-Bugzilla: 1968519
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>
RH-Acked-by: Thomas Huth <thuth@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>

Revert pc_rom_ro hack.
Remove force_rev1_fadt hack, it is not needed anymore.

Signed-off-by: Juan Quintela <quintela@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/acpi/ich9.c         | 15 ---------------
 hw/core/machine.c      |  9 ---------
 hw/i386/acpi-build.c   |  3 ---
 hw/i386/pc.c           | 18 +-----------------
 hw/i386/pc_piix.c      | 19 -------------------
 hw/i386/pc_q35.c       | 18 ------------------
 hw/ppc/spapr.c         | 43 ------------------------------------------
 include/hw/acpi/ich9.h |  3 ---
 include/hw/boards.h    |  3 ---
 include/hw/i386/pc.h   |  6 ------
 10 files changed, 1 insertion(+), 136 deletions(-)

diff --git a/hw/acpi/ich9.c b/hw/acpi/ich9.c
index 33b0c6e33c..7f01fad64c 100644
--- a/hw/acpi/ich9.c
+++ b/hw/acpi/ich9.c
@@ -369,18 +369,6 @@ static void ich9_pm_set_enable_tco(Object *obj, bool value, Error **errp)
     s->pm.enable_tco = value;
 }
 
-static bool ich9_pm_get_force_rev1_fadt(Object *obj, Error **errp)
-{
-    ICH9LPCState *s = ICH9_LPC_DEVICE(obj);
-    return s->pm.force_rev1_fadt;
-}
-
-static void ich9_pm_set_force_rev1_fadt(Object *obj, bool value, Error **errp)
-{
-    ICH9LPCState *s = ICH9_LPC_DEVICE(obj);
-    s->pm.force_rev1_fadt = value;
-}
-
 void ich9_pm_add_properties(Object *obj, ICH9LPCPMRegs *pm)
 {
     static const uint32_t gpe0_len = ICH9_PMIO_GPE0_LEN;
@@ -403,9 +391,6 @@ void ich9_pm_add_properties(Object *obj, ICH9LPCPMRegs *pm)
     object_property_add_bool(obj, "cpu-hotplug-legacy",
                              ich9_pm_get_cpu_hotplug_legacy,
                              ich9_pm_set_cpu_hotplug_legacy);
-    object_property_add_bool(obj, "__com.redhat_force-rev1-fadt",
-                             ich9_pm_get_force_rev1_fadt,
-                             ich9_pm_set_force_rev1_fadt);
     object_property_add_uint8_ptr(obj, ACPI_PM_PROP_S3_DISABLED,
                                   &pm->disable_s3, OBJ_PROP_FLAG_READWRITE);
     object_property_add_uint8_ptr(obj, ACPI_PM_PROP_S4_DISABLED,
diff --git a/hw/core/machine.c b/hw/core/machine.c
index 0bcaabd8a9..6dc2651d73 100644
--- a/hw/core/machine.c
+++ b/hw/core/machine.c
@@ -215,15 +215,6 @@ GlobalProperty hw_compat_rhel_7_5[] = {
 };
 const size_t hw_compat_rhel_7_5_len = G_N_ELEMENTS(hw_compat_rhel_7_5);
 
-/* Mostly like hw_compat_2_9 except
- * x-mtu-bypass-backend, x-migrate-msix has already been
- * backported to RHEL7.4. shpc was already on in 7.4.
- */
-GlobalProperty hw_compat_rhel_7_4[] = {
-    { "intel-iommu", "pt", "off" },
-};
-
-const size_t hw_compat_rhel_7_4_len = G_N_ELEMENTS(hw_compat_rhel_7_4);
 GlobalProperty hw_compat_5_2[] = {
     { "ICH9-LPC", "smm-compat", "on"},
     { "PIIX4_PM", "smm-compat", "on"},
diff --git a/hw/i386/acpi-build.c b/hw/i386/acpi-build.c
index 7bd67f7877..de98750aef 100644
--- a/hw/i386/acpi-build.c
+++ b/hw/i386/acpi-build.c
@@ -231,9 +231,6 @@ static void acpi_get_pm_info(MachineState *machine, AcpiPmInfo *pm)
         pm->fadt.reset_reg = r;
         pm->fadt.reset_val = 0xf;
         pm->fadt.flags |= 1 << ACPI_FADT_F_RESET_REG_SUP;
-        if (object_property_get_bool(lpc,
-                                     "__com.redhat_force-rev1-fadt", NULL))
-            pm->fadt.rev = 1;
         pm->cpu_hp_io_base = ICH9_CPU_HOTPLUG_IO_BASE;
         pm->smi_on_cpuhp =
             !!(smi_features & BIT_ULL(ICH9_LPC_SMI_F_CPU_HOTPLUG_BIT));
diff --git a/hw/i386/pc.c b/hw/i386/pc.c
index 75abe0acc2..6b39abce02 100644
--- a/hw/i386/pc.c
+++ b/hw/i386/pc.c
@@ -477,20 +477,6 @@ GlobalProperty pc_rhel_7_5_compat[] = {
 };
 const size_t pc_rhel_7_5_compat_len = G_N_ELEMENTS(pc_rhel_7_5_compat);
 
-GlobalProperty pc_rhel_7_4_compat[] = {
-    /* pc_rhel_7_4_compat from pc_compat_2_9 */ 
-    { "mch", "extended-tseg-mbytes", stringify(0) },
-    /* bz 1489800 */ 
-    { "ICH9-LPC", "__com.redhat_force-rev1-fadt", "on" },
-    /* pc_rhel_7_4_compat from pc_compat_2_10 */ 
-    { "i440FX-pcihost", "x-pci-hole64-fix", "off" },
-    /* pc_rhel_7_4_compat from pc_compat_2_10 */ 
-    { "q35-pcihost", "x-pci-hole64-fix", "off" },
-    /* pc_rhel_7_4_compat from pc_compat_2_10 */ 
-    { TYPE_X86_CPU, "x-hv-max-vps", "0x40" },
-};
-const size_t pc_rhel_7_4_compat_len = G_N_ELEMENTS(pc_rhel_7_4_compat);
-
 /*
  * The PC_RHEL_*_COMPAT serve the same purpose for RHEL-7 machine
  * types as the PC_COMPAT_* do for upstream types.
@@ -1094,8 +1080,7 @@ void pc_memory_init(PCMachineState *pcms,
     option_rom_mr = g_malloc(sizeof(*option_rom_mr));
     memory_region_init_ram(option_rom_mr, NULL, "pc.rom", PC_ROM_SIZE,
                            &error_fatal);
-    /* RH difference: See bz 1489800, explicitly make ROM ro */
-    if (pcmc->pc_rom_ro) {
+    if (pcmc->pci_enabled) {
         memory_region_set_readonly(option_rom_mr, true);
     }
     memory_region_add_subregion_overlap(rom_memory,
@@ -1845,7 +1830,6 @@ static void pc_machine_class_init(ObjectClass *oc, void *data)
     pcmc->pvh_enabled = true;
     pcmc->kvmclock_create_always = true;
     assert(!mc->get_hotplug_handler);
-    pcmc->pc_rom_ro = true;
     mc->async_pf_vmexit_disable = false;
     mc->get_hotplug_handler = pc_get_hotplug_handler;
     mc->hotplug_allowed = pc_hotplug_allowed;
diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index 64662cc3d5..fe2ac7593a 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -1011,22 +1011,3 @@ static void pc_machine_rhel750_options(MachineClass *m)
 
 DEFINE_PC_MACHINE(rhel750, "pc-i440fx-rhel7.5.0", pc_init_rhel750,
                   pc_machine_rhel750_options);
-
-static void pc_init_rhel740(MachineState *machine)
-{
-    pc_init1(machine, TYPE_I440FX_PCI_HOST_BRIDGE, \
-             TYPE_I440FX_PCI_DEVICE);
-}
-
-static void pc_machine_rhel740_options(MachineClass *m)
-{
-    PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
-    pc_machine_rhel750_options(m);
-    m->desc = "RHEL 7.4.0 PC (i440FX + PIIX, 1996)";
-    pcmc->pc_rom_ro = false;
-    compat_props_add(m->compat_props, hw_compat_rhel_7_4, hw_compat_rhel_7_4_len);
-    compat_props_add(m->compat_props, pc_rhel_7_4_compat, pc_rhel_7_4_compat_len);
-}
-
-DEFINE_PC_MACHINE(rhel740, "pc-i440fx-rhel7.4.0", pc_init_rhel740,
-                  pc_machine_rhel740_options);
diff --git a/hw/i386/pc_q35.c b/hw/i386/pc_q35.c
index bf49a943dc..9958ed9184 100644
--- a/hw/i386/pc_q35.c
+++ b/hw/i386/pc_q35.c
@@ -774,21 +774,3 @@ static void pc_q35_machine_rhel750_options(MachineClass *m)
 
 DEFINE_PC_MACHINE(q35_rhel750, "pc-q35-rhel7.5.0", pc_q35_init_rhel750,
                   pc_q35_machine_rhel750_options);
-
-static void pc_q35_init_rhel740(MachineState *machine)
-{
-    pc_q35_init(machine);
-}
-
-static void pc_q35_machine_rhel740_options(MachineClass *m)
-{
-    PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
-    pc_q35_machine_rhel750_options(m);
-    m->desc = "RHEL-7.4.0 PC (Q35 + ICH9, 2009)";
-    pcmc->pc_rom_ro = false;
-    compat_props_add(m->compat_props, hw_compat_rhel_7_4, hw_compat_rhel_7_4_len);
-    compat_props_add(m->compat_props, pc_rhel_7_4_compat, pc_rhel_7_4_compat_len);
-}
-
-DEFINE_PC_MACHINE(q35_rhel740, "pc-q35-rhel7.4.0", pc_q35_init_rhel740,
-                  pc_q35_machine_rhel740_options);
diff --git a/hw/ppc/spapr.c b/hw/ppc/spapr.c
index ca0b99403e..bdabbf8b81 100644
--- a/hw/ppc/spapr.c
+++ b/hw/ppc/spapr.c
@@ -5308,49 +5308,6 @@ static void spapr_machine_rhel750sxxm_class_options(MachineClass *mc)
 }
 
 DEFINE_SPAPR_MACHINE(rhel750sxxm, "rhel7.5.0-sxxm", false);
-
-/*
- * pseries-rhel7.4.0
- * like spapr_compat_2_9
- */
-GlobalProperty spapr_compat_rhel7_4[] = {
-    { TYPE_POWERPC_CPU, "pre-2.10-migration", "on" },
-};
-const size_t spapr_compat_rhel7_4_len = G_N_ELEMENTS(spapr_compat_rhel7_4);
-
-static void spapr_machine_rhel740_class_options(MachineClass *mc)
-{
-    SpaprMachineClass *smc = SPAPR_MACHINE_CLASS(mc);
-
-    spapr_machine_rhel750_class_options(mc);
-    compat_props_add(mc->compat_props, hw_compat_rhel_7_4, hw_compat_rhel_7_4_len);
-    compat_props_add(mc->compat_props, spapr_compat_rhel7_4, spapr_compat_rhel7_4_len);
-    smc->has_power9_support = false;
-    smc->pre_2_10_has_unused_icps = true;
-    smc->resize_hpt_default = SPAPR_RESIZE_HPT_DISABLED;
-    smc->default_caps.caps[SPAPR_CAP_HTM] = SPAPR_CAP_ON;
-}
-
-DEFINE_SPAPR_MACHINE(rhel740, "rhel7.4.0", false);
-
-/*
- * pseries-rhel7.4.0-sxxm
- *
- * pseries-rhel7.4.0 with speculative execution exploit mitigations enabled by default
- */
-
-static void spapr_machine_rhel740sxxm_class_options(MachineClass *mc)
-{
-    SpaprMachineClass *smc = SPAPR_MACHINE_CLASS(mc);
-
-    spapr_machine_rhel740_class_options(mc);
-    smc->default_caps.caps[SPAPR_CAP_CFPC] = SPAPR_CAP_WORKAROUND;
-    smc->default_caps.caps[SPAPR_CAP_SBBC] = SPAPR_CAP_WORKAROUND;
-    smc->default_caps.caps[SPAPR_CAP_IBS] = SPAPR_CAP_FIXED_CCD;
-}
-
-DEFINE_SPAPR_MACHINE(rhel740sxxm, "rhel7.4.0-sxxm", false);
-
 static void spapr_machine_register_types(void)
 {
     type_register_static(&spapr_machine_info);
diff --git a/include/hw/acpi/ich9.h b/include/hw/acpi/ich9.h
index e1ecfbaf1f..df519e40b5 100644
--- a/include/hw/acpi/ich9.h
+++ b/include/hw/acpi/ich9.h
@@ -62,9 +62,6 @@ typedef struct ICH9LPCPMRegs {
     bool smm_compat;
     bool enable_tco;
     TCOIORegs tco_regs;
-
-    /* RH addition, see bz 1489800 */
-    bool force_rev1_fadt;
 } ICH9LPCPMRegs;
 
 #define ACPI_PM_PROP_TCO_ENABLED "enable_tco"
diff --git a/include/hw/boards.h b/include/hw/boards.h
index 24d4d726d8..f27170b6fb 100644
--- a/include/hw/boards.h
+++ b/include/hw/boards.h
@@ -436,7 +436,4 @@ extern const size_t hw_compat_rhel_7_6_len;
 extern GlobalProperty hw_compat_rhel_7_5[];
 extern const size_t hw_compat_rhel_7_5_len;
 
-extern GlobalProperty hw_compat_rhel_7_4[];
-extern const size_t hw_compat_rhel_7_4_len;
-
 #endif
diff --git a/include/hw/i386/pc.h b/include/hw/i386/pc.h
index 820fb3f577..ff93dfb372 100644
--- a/include/hw/i386/pc.h
+++ b/include/hw/i386/pc.h
@@ -128,9 +128,6 @@ struct PCMachineClass {
 
     /* create kvmclock device even when KVM PV features are not exposed */
     bool kvmclock_create_always;
-
-    /* RH only, see bz 1489800 */
-    bool pc_rom_ro;
 };
 
 #define TYPE_PC_MACHINE "generic-pc-machine"
@@ -302,9 +299,6 @@ extern const size_t pc_rhel_7_6_compat_len;
 extern GlobalProperty pc_rhel_7_5_compat[];
 extern const size_t pc_rhel_7_5_compat_len;
 
-extern GlobalProperty pc_rhel_7_4_compat[];
-extern const size_t pc_rhel_7_4_compat_len;
-
 /* Helper for setting model-id for CPU models that changed model-id
  * depending on QEMU versions up to QEMU 2.4.
  */
-- 
2.27.0

